src/node_buffer.cc

```diff
@@ -1072,6 +1072,42 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
                                 : -1);
 }
 
+void SwapShort(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
+  SPREAD_ARG(args.This(), ts_obj);
+
+  if (ts_obj_length % 2 != 0)
+    return env->ThrowRangeError(
+      "Buffer length must be a multiple of 16-bits");
+
+  for (size_t i = 0; i < ts_obj_length; i += 2) {
+    const uint8_t lo = static_cast<uint8_t>(ts_obj_data[i + 0]);
+    const uint8_t hi = static_cast<uint8_t>(ts_obj_data[i + 1]);
+    ts_obj_data[i + 0] = hi;
+    ts_obj_data[i + 1] = lo;
+  }
+  args.GetReturnValue().Set(args.This());
+}
+
+void SwapLong(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
+  SPREAD_ARG(args.This(), ts_obj);
+
+  if (ts_obj_length % 4 != 0)
+    return env->ThrowRangeError(
+      "Buffer length must be a multiple of 32-bits");
+
+  uint32_t* data = reinterpret_cast<uint32_t*>(ts_obj_data);
```

Comment:
I meant byte aligned memory address. Say:
```js
const b = Buffer(16);  // address 0x30b2c40
b.swap32();
const s = b.slice(1);  // address 0x30b2c41
b.swap32();
```
The cast operation need to happen on aligned memory, I believed based on the byte length of the operation performed. So in this case it would need to be 4-byte aligned.

There was a bug in v8 causing us trouble a while back for this very reason. It was fixed in 36ac3d642e. We also had this issue with the Buffer pool 43660ad3. And for a while ucs2 was broken if attempting to write to unaligned memory 68e53ddc.